pkg/controller/batch/computeenvironment/setup.go

```diff
@@ -0,0 +1,553 @@
+		//allocationStrategy, bidPercentage, ec2Configuration, ec2KeyPair, imageId, instanceRole, instanceTypes,
+		//launchTemplate, placementGroup, securityGroupIds, subnets, tags, type, updatePolicy,
+		//updateToLatestImageVersion can be updated for CE only with Allocation Strategy BEST_FIT_PROGRESSIVE and SPOT_CAPACITY_OPTIMIZED
+		//->this + the modifiable field in AWS Console suggest that updating/changing the spotfleetrole should be possible,
+		//however 1. update does not care/have that field 2. changing it in AWS console seems to also not get updated correctly...(+buggy dropdown?)
+		//->so we just ignore it here!
+
+		//TODO: test tags (CE, CR) for changes
+		// CE Tags changes: "These tags can be updated or removed using the TagResource and UntagResource API operations."
+		// CR Tags changes: These tags can't be updated or removed after the compute environment is created.
+		// Any changes to these tags require that you create a new compute environment and remove the old compute environment.
+		return false, nil
+	}
+
+	fmt.Println("isUpToDate true: aka everything fine")
+	return true, nil
+}
+
+func lateInitialize(spec *svcapitypes.ComputeEnvironmentParameters, resp *svcsdk.DescribeComputeEnvironmentsOutput) error {
+	fmt.Println("lateInitialize")
+
+	ce := resp.ComputeEnvironments[0]
+	fmt.Printf("lateInitialize: ce: %v\n", ce)
+	spec.DesiredState = awsclients.LateInitializeStringPtr(spec.DesiredState, ce.State)
+	fmt.Printf("lateInitialize: spec.DesiredState: %v\n", awsclients.StringValue(spec.DesiredState))
+
+	fmt.Printf("lateInitialize spec.ServiceRoleARN before possibly setting it here: %v\n", awsclients.StringValue(spec.ServiceRoleARN))
+	fmt.Printf("lateInitialize spec.ServiceRoleRef: %v\n", spec.ServiceRoleARNRef)
+
+	//for when no role provided by user, we can use an autocreated/default service-linked-role by AWS (only type MANAGED)
+	//i shouldnt need that AWS-made role in my specs, bc i cant change it, no?
+	//also for the lateinit of the objspces from isuptodate needed...so we can compare
+	//in case CEType = MANAGED and user did not provide a ServiceRole, AWS will auto-assign/-create a Service-Linked-Role for Batch
+	//(setting the spec to it, for isuptodate comparing compatibility)
+	if spec.ServiceRoleARN == nil {
+		fmt.Println("lateInitialize: spec.ServiceRole == nil, setting it to ce.ServiceRole")
+		spec.ServiceRoleARN = ce.ServiceRole
+		// if not provided, CE of type MANAGED does get automatically given a Service-Linked-Role for Batch by aws (either an existing or a newly created one)
+	}
+
+	//does this work with fargate???
+	//seems to work fine with FARGATE (and FARGATESPOT), sets them to 0
+	if ce.ComputeResources != nil {
+		fmt.Println("lateInitialize: if ce.ComputeResources != nil ")
+		spec.ComputeResources.DesiredvCPUs = ce.ComputeResources.DesiredvCpus // overwrite what the user put in to what AWS sets it to (seems to be heavily AWS managed)
```

Comment:
I think we should not overwrite the field if its is set by the user. If a diff occurs between what the spec and what AWS wants then we need to resolve somehow else.